internal/net/httpserver.go

```diff
@@ -57,40 +66,38 @@ func (h *HTTPServer) Listener() net.Listener {
 // An error is returned if the server failed to start up, if the server was
 // already listening, or if the server was stopped with Stop().
 func (h *HTTPServer) ListenAndServe() error {
-	h.lock.Lock()
-	defer h.lock.Unlock()
-
-	if atomic.LoadUint32(&h.stopped) == 1 {
-		return errors.New("the server has been stopped")
-	}
-
-	if h.listener != nil {
-		return errors.New("the server is already listening")
+	if h.stopped.Load() {
+		return errServerStopped
 	}
 
 	addr := h.Server.Addr
```

Comment:
Ah, my bad. We don't mutate `h.Server` ever so this should be fine, but I'll
move it below the lock anyway.